R/dist_student_t.R

```diff
@@ -1,26 +1,30 @@
-#' The (non-central) Student t Distribution
+#' The (non-central) location-scale Student t Distribution
 #'
 #' \lifecycle{stable}
 #'
 #' @inheritParams stats::dt
+#' @param mu The location parameter of the distribution.
+#'   If `ncp == 0` (or `NULL`), this is the median.
+#' @param sigma The scale parameter of the distribution.
 #'
 #' @seealso [stats::TDist]
 #'
 #' @examples
-#' dist_student_t(df = c(1,2,5))
+#' dist_student_t(df = c(1,2,5), mu = c(0,1,2), sigma = c(1,2,3))
 #'
 #' @name dist_student_t
 #' @export
-dist_student_t <- function(df, ncp = NULL){
+dist_student_t <- function(df, mu = 0, sigma = 1, ncp = NULL){
   df <- vec_cast(df, if(any(is.infinite(df))) numeric() else integer())
   if(any(df <= 0)){
     abort("The degrees of freedom parameter of a Student t distribution must be strictly positive.")
   }
-  if(is.null(ncp)){
-    new_dist(df = df, class = "dist_student_t")
-  } else {
-    new_dist(df = df, ncp = ncp, class = "dist_student_t")
+  mu <- vec_cast(mu, double())
+  sigma <- vec_cast(sigma, double())
+  if(any(sigma[!is.na(sigma)] <= 0)){
+    abort("The scale parameter of a Student t distribution must be strictly positive.")
```

Comment:
Could this warning specify that the scale parameter is `sigma`?
Perhaps `"The scale (sigma) parameter of ..."`

Reply:
should be fixed now, thanks!